src/main/java/at/ac/tuwien/kr/alpha/common/terms/VariableTerm.java

```diff
@@ -23,6 +48,10 @@ private VariableTerm(String variableName) {
 	}
 
 	public static VariableTerm getInstance(String variableName) {
+		final char firstChar = variableName.charAt(0);
+		if (firstChar != ANONYMOUS_VARIABLE_PREFIX.charAt(0) && !Character.isUpperCase(firstChar)) {
+			throw new IllegalArgumentException("Variable name must start with upper-case letter or " + ANONYMOUS_VARIABLE_PREFIX);
+		}
```

Comment:
I'm assuming this can only ever be triggered from unit tests (since anything triggering the `if` would be taken care of by the parser normally), so I'm not sure this check should be here. At least, I think there should be a comment explaining it's purpose - for someone "naively" reading the code it might look like a syntax check for parsed programs otherwise..

Alternatively, would your problem be solved by writing a utlity method for unit tests? something like the below snippet, that can be used in tests, but does not "bloat" the actual production code:
```
public static VariableTerm generateVariableTerm(String varName){
    if(!varName.maches("(_)|([A-Z][A-Za-z0-9_]*))){
        throw .....
    }
    return new VariableTerm(varName);
}